Scripts/Objects/Unit.gd

```diff
@@ -25,39 +18,44 @@ var cost = 0
 var defense = 1
 var offense = 1
 var speed = 1
-# TODO: Think about how we want store stats at the data layer
-# need second pass on design to be more exact
+
+var state = 0 #0 = disabled, 1 = can move and attack, 2 = can attack, not move, 3 = can attack, not move
```

Comment:
we can do `enum { DISABLED, CAN_MOVE_AND_ATTACK, etc }` right? Would rather have these things compiler-enforced if possible.

Reply:
it's interpreted language so no compiling. But yes we can change it, I tried doing it that way at first but I couldn't figure out the syntax